src/controller/AuthController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+require_once __DIR__ . '/../model/UserModel.php';
+class AuthController {
+    private $userModel;
+
+    public function __construct() {
+        global $conn;
+        $this->userModel = new UserModel($conn);
+    }
+
+    public function index() {
+        if ($_SERVER["REQUEST_METHOD"] == "POST") {
+            if (isset($_POST['login_submit'])) {
+                $this->handleLogin();
+            } elseif (isset($_POST['register_submit'])) {
+                $this->handleRegistration();
+            }
+        } else {
+            $this->render('login', ['login_error' => '']);
+        }
+    }
+
+    private function handleLogin() {
+        session_start();
+        $username_email = $_POST['username_email'];
+        $password = $_POST['password'];
+
+        $user = $this->userModel->loginUser($username_email, $password);
+
+        if ($user) {
+            $_SESSION['user_id'] = $user['id'];
+            $_SESSION['username'] = $user['username'];
+            header("Location: home");
+            exit();
+        } else {
+            $login_error = "Invalid username or password";
+            $this->render('login', ['login_error' => $login_error]);
+        }
+    }
+
+    private function handleRegistration() {
+        $username = $_POST['register_username'];
+        $email = $_POST['register_email'];
+        $password = $_POST['register_password'];
+    
+        if ($this->userModel->getUserByUsername($username)) {
+            $register_error = "Username already taken. Choose a different username.";
```

Comment:
I understand what you are doing here, but in these types of situations, It's better to throw and catch exceptions. We could throw the exception in the model and then catch it on the controller. So we have a centralized way of dealing with the exception.